### ISSUE TITLE
[FEATURE] Volume: Support logarithmic gain ramps

### ISSUE THREAD

**Issue by singalsu** [thesofproject/sof#1386]
Logaritmic gain ramps create subjectively equal amount of loudness change per unit of time so it could be the preferred ramp type. It also allows usage of shorter time ramps while avoiding clicks or zipper noise. Currently SOF only supports linear ramps. The topology and driver supports the ramp type topology token so it would be a FW only change.

The ramp type was omitted from recent PR for volume ramp. The implementation would need log() function to DSP. Such function to cover the volume range with is possible to compute via a Taylor series approximation/iteration done with integer fractional math.

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by lgirdwood** [comment]
@singalsu I dont think we need the precision of log() in the DSP for gain ramping, we most likely need a compressed log lookup table. i.e. log table uses same array element where log() output does not significantly change compared to previous or next input values. iiuc, we should only need to support a finite and known range of possible log() input values too ?

**Comment by kv2019i** [comment]
@singalsu No activity in 5 years, still relevant?

**Comment by singalsu** [comment]
I think we can close this, we have the Windows Fade as a more sophisticated option over basic linear ramp.
